### PR TITLE
Modular etherspot wallet

### PR DESCRIPTION
**Pull request by lbw33**
## Description

- Removal of ERC7579 and ERC6900 submodules and import of ERC7579 contracts required.
- Changed naming from `EtherspotWallet7579` to `ModularEtherspotWallet`.
- Rework of wallet, factory and base wallet to comply with updated standard.
- Updating of test suite with utilisation of standard tests.
- Addition of testing utils to make test setup easier.
- Update of deployment scripting.
- Updating gas estimation documents.

- Updated hardhat.config to abstract solidity compliers, chain configs and Etherscan configs into their own separate config files `/config`.
- Added setup script command to make setup easier and updated `README.md`.

## Motivation and Context

- To comply with updated ERC7579 standard for Modular Smart Accounts.

## How Has This Been Tested?

- Test suite (testing code).
- Local deployment (testing deploy script).

## Screenshots (if appropriate):
**Test Suite:**
Screenshot 2024-03-01 at 16 51 22

**Successful Local Deployment:**
Screenshot 2024-02-26 at 14 42 39

**Gas Estimations**
Screenshot 2024-03-01 at 16 50 16

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by ch4r10t33r** [comment]
@lbw33 Pls resolve merge conflicts and issues reported by openzeppelin inspector.

File: src/modular-etherspot-wallet/wallet/ModularEtherspotWallet.sol [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+     */
+    function accountId()
+        external
+        view
+        virtual
+        override
+        returns (string memory)
+    {
+        // vendor.flavour.semver
+        return "uMSA.simple/noHook.v0.1";
```
Comment (ch4r10t33r) [inline src/modular-etherspot-wallet/wallet/ModularEtherspotWallet.sol:213]: Shouldn't this read "Etherspot/v0.1"?

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]
Also pls delete old stale branches from the repo

**Comment by arddluma** [comment]
Deleting comment from openzeppelin-code as it contains false positives.

File: src/modular-etherspot-wallet/wallet/ModularEtherspotWallet.sol (hunk 1 above)
Comment (lbw33) [reply src/modular-etherspot-wallet/wallet/ModularEtherspotWallet.sol:213]: Sorry yes, amended to "ModularEtherspotWallet/.v0.1"

File: src/modular-etherspot-wallet/wallet/ModularEtherspotWalletFactory.sol [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+contract ModularEtherspotWalletFactory {
     address public immutable implementation;
 
-    constructor(address _ewImplementation) {
-        implementation = _ewImplementation;
+    constructor(address _msaImplementation) {
+        implementation = _msaImplementation;
     }
 
     function createAccount(
```
[the remark this answers is not in the document]
Comment (lbw33) [reply src/modular-etherspot-wallet/wallet/ModularEtherspotWalletFactory.sol:14]: We shouldn't need to include an offset as the initcode will always be different as it contains the EOA address:

```
bytes memory _initCode = abi.encode(
            **owner1**,
            address(bootstrapSingleton),
            abi.encodeCall(
                bootstrapSingleton.initMSA,
                (validators, executors, hook, fallbackHandler)
            )
        );
```

**Review by ch4r10t33r — APPROVED** [review]

**Review by 0xSulpiride — APPROVED** [review]
LGTM

**Pull request merged**